server/api/utils/singletons/k8s.py

```diff
@@ -591,6 +591,7 @@ def __init__(
         project=None,
         default_pod_spec_attributes=None,
         resources=None,
+        labels={},
```

Comment:
value is mutable, change to `None` and set within function if its value is None

Reply:
Good thinking. I've improved it for us. Please see if it makes more sense now.